File: java/media_idades.java
import java.util.Locale;
import java.util.Scanner;

public class media_idades {

	public static void main(String[] args) {
		
		Locale.setDefault(Locale.US);
		Scanner sc = new Scanner(System.in);
		
		int idades, cont;
		double media, somaIdades;
		
		System.out.println("Digite as idades: ");
		idades = sc.nextInt();
		
		cont = 0;
		somaIdades = 0;
			while (idades > 0) {
				cont = cont + 1;
				somaIdades = somaIdades + idades;
				idades = sc.nextInt();
			}
		
		if (cont == 0) {
			System.out.println("IMPOSSIVEL CALCULAR");
		} else {	
			media = somaIdades / cont; 
			System.out.println("MEDIA = " + String.format("%.2f", media));	
		}
		
		sc.close();
	}

}

File: c++/media_idades.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {

    int idade, cont = 0, somaIdades = 0;
    double media;

    cout << "Digite as idades:" << endl;
    cin >> idade;

    if (idade < 0 ) {
        cout << "IMPOSSIVEL CALCULAR" << endl;
    } else {
        while (idade > 0) {
                somaIdades = somaIdades + idade;
                cont = cont + 1;
                cin >> idade;
    }

        media = (double) somaIdades / cont;

        cout << fixed << setprecision(2);
        cout << "MEDIA = " << media << endl;
    }

    return 0;
}
